subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_pdu.h

```diff
@@ -65,6 +65,26 @@ static inline struct pdu_adv *lll_adv_scan_rsp_peek(struct lll_adv *lll)
 	return (void *)lll->scan_rsp.pdu[lll->scan_rsp.last];
 }
 
+static inline struct pdu_adv *lll_adv_pdu_latest_peek(struct lll_adv_pdu *pdu)
+{
+	uint8_t first;
+
+	first = pdu->first;
+	if (first != pdu->last) {
+		first += 1U;
```

Comment:
Just curious: Is there a reason why you are using `+= 1U` instead of `first++`?

Reply:
> ;-) or `++first` (in C it does not matter, but following general conventions this is the version of incrementation that should be used in situations `x = x + 1`)

That's an old recommendation for old compilers though, right? 
IIRC a long time ago, `++i` and `i++` would result in different code, where `++i` would use one less register or something like that, but I think it's a long time since any reasonable compiler would do this.

Reply:
Yes that is correct. I'm used to go for the pre-incrementation in situations like this.

Reply:
The only different I think of, is the size of `first` which is `uint8_t`, and by using `first += 1U;` the compiler has the choice to use a byte instead of a word to add LSB of the registers. Use of unary operator makes sense when the variable is machine word size, in which case an inc instruction will be used.

```
    uint8_t x = 5;
    
    x = x + 1;
```

versus

```
    int i;
    
    i++;
```

But I am not sure, what actually the compiler is doing.